pkg/auth/authLogin.go

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package auth
+
+import (
+    "context"
+    "encoding/json"
+    "log"
+    "path/filepath"
+
+    "github.com/galasa.dev/cli/pkg/api"
+    galasaErrors "github.com/galasa.dev/cli/pkg/errors"
+    "github.com/galasa.dev/cli/pkg/files"
+    "github.com/galasa.dev/cli/pkg/galasaapi"
+    "github.com/galasa.dev/cli/pkg/props"
+    "github.com/galasa.dev/cli/pkg/utils"
+)
+
+// Login - performs all the logic to implement the `galasactl auth login` command
+func Login(apiServerUrl string, fileSystem files.FileSystem, galasaHome utils.GalasaHome) error {
+
+    var err error = nil
+    var authProperties galasaapi.AuthProperties
+    authProperties, err = getAuthProperties(fileSystem, galasaHome)
+    if err == nil {
+        var jwt string
+        jwt, err = GetJwtFromRestApi(apiServerUrl, authProperties)
+        if err == nil {
+            err = writeBearerTokenJsonFile(fileSystem, galasaHome, jwt)
+        }
+    }
+
+    return err
+}
+
+// Gets authentication properties from the user's galasactl.properties file
+func getAuthProperties(fileSystem files.FileSystem, galasaHome utils.GalasaHome) (galasaapi.AuthProperties, error) {
+    var err error = nil
+    authProperties := galasaapi.NewAuthProperties()
+
+    galasactlPropertiesFilePath := filepath.Join(galasaHome.GetNativeFolderPath(), "galasactl.properties")
+    galasactlProperties, err := props.ReadPropertiesFile(fileSystem, galasactlPropertiesFilePath)
+    if err == nil {
+        clientIdProperty := "GALASA_CLIENT_ID"
```

Comment:
Can we separate these auth properties into a separate file, so symbolic constants can be used from anywhere in the code ?

Something like authProperties ?

Then the getAuthPropertyList() could return this slice of property names ?

Reply:
The validate of the properties could also be moved there perhaps.

Reply:
Addressed in https://github.com/galasa-dev/cli/pull/151/commits/a3679e637189ed9792675899842f9067b570bf06